gsa/src/web/pages/scanners/dialog.js

```diff
@@ -140,211 +165,233 @@ class ScannerDialog extends React.Component {
+  };
+
+  const data = {
+    ca_pub,
+    comment,
+    host,
+    id,
+    name,
+    port,
+    which_cert,
+  };
+
+  let SCANNER_TYPES;
+
+  const {gmp} = props;
+
+  if (gmp.settings.enableGreenboneSensor) {
+    SCANNER_TYPES = [
+      GMP_SCANNER_TYPE,
+      OSP_SCANNER_TYPE,
+      GREENBONE_SENSOR_SCANNER_TYPE,
+    ];
+  } else {
+    SCANNER_TYPES = [GMP_SCANNER_TYPE, OSP_SCANNER_TYPE];
   }
 
-  render() {
-    const {
-      comment = '',
-      gmp,
-      scanner,
-      ca_pub,
-      credentials,
-      host = 'localhost',
-      id,
-      name = _('Unnamed'),
-      port = '9391',
-      title = _('New Scanner'),
-      type = OSP_SCANNER_TYPE,
-      which_cert,
-      onClose,
-      onCredentialChange,
-      onNewCredentialClick,
-      onSave,
-    } = this.props;
-
-    let SCANNER_TYPES;
-
-    if (gmp.settings.enableGreenboneSensor) {
-      SCANNER_TYPES = [
-        GMP_SCANNER_TYPE,
-        OSP_SCANNER_TYPE,
-        GREENBONE_SENSOR_SCANNER_TYPE,
-      ];
-    } else {
-      SCANNER_TYPES = [GMP_SCANNER_TYPE, OSP_SCANNER_TYPE];
-    }
+  let {credential_id} = props;
 
-    let {credential_id} = this.props;
-
-    const data = {
-      ca_pub,
-      comment,
-      host,
-      id,
-      name,
-      port,
-      which_cert,
-    };
-
-    const scannerTypesOptions = map(SCANNER_TYPES, scannerType => ({
-      label: scannerTypeName(scannerType),
-      value: scannerType,
-    }));
-
-    const scanner_credentials = filter_credentials(credentials, type);
-    const is_edit = isDefined(scanner);
-    const isInUse = isDefined(scanner) && scanner.isInUse();
-    const show_cred_info =
-      isDefined(scanner) &&
-      isDefined(scanner.credential) &&
-      scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
-
-    const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
-    const isOspScannerType = type === OSP_SCANNER_TYPE;
-
-    if (isGreenboneSensorType) {
-      credential_id = '';
-    }
+  const scannerTypesOptions = map(SCANNER_TYPES, scannerType => ({
+    label: scannerTypeName(scannerType),
+    value: scannerType,
+  }));
+
+  const scanner_credentials = filter_credentials(credentials, type);
+  const is_edit = isDefined(scanner);
+  const isInUse = isDefined(scanner) && scanner.isInUse();
+  const show_cred_info =
+    isDefined(scanner) &&
+    isDefined(scanner.credential) &&
+    scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
+
+  const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
+  const isOspScannerType = type === OSP_SCANNER_TYPE;
+
+  if (isGreenboneSensorType) {
+    credential_id = '';
+  }
+
+  const [activeTab, setActiveTab] = useState(0);
+
+  const totalPages = 2;
+
+  const {page, handlePageTurn, prevDisabled, nextDisabled} = usePageTurn({
+    activeTab,
+    lastPage: totalPages - 1,
+  });
+
+  useEffect(() => {
+    setActiveTab(page);
+  }, [page]);
+
+  return (
+    <SaveDialog
```

Comment:
A small nitpick: Could you sort the props by alphabet and put the on... handlers after every other prop?